tests/rptest/services/openmessaging_benchmark.py

```diff
@@ -0,0 +1,276 @@
+# Copyright 2020 Vectorized, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import re
+import threading
+import os
+import json
+import collections
+
+from ducktape.services.service import Service
+from ducktape.utils.util import wait_until
+from rptest.services.utils import BadLogLines, NodeCrash
+
+
+# Benchmark worker that is used by benchmark process to run consumers and producers
+class OpenMessagingBenchmarkWorkers(Service):
+    PERSISTENT_ROOT = "/var/lib/openmessaging"
+    STDOUT_STDERR_CAPTURE = os.path.join(PERSISTENT_ROOT, "worker.log")
+    PORT = 9090
+    STATS_PORT = 9091
+
+    logs = {
+        "open_messaging_benchmark_worker_stdout_stderr": {
+            "path": STDOUT_STDERR_CAPTURE,
+            "collect_default": True
+        }
+    }
+
+    def __init__(self, ctx, num_workers=3):
+        super(OpenMessagingBenchmarkWorkers,
+              self).__init__(ctx, num_nodes=num_workers)
+
+    def start_node(self, node):
+        self.logger.info("Starting Open Messaging Benchmark worker node on %s",
+                         node.account.hostname)
+
+        self.clean_node(node)
+
+        node.account.mkdirs(OpenMessagingBenchmarkWorkers.PERSISTENT_ROOT)
+
+        start_cmd = f"cd /opt/openmessaging-benchmark; \
+                      bin/benchmark-worker \
+                      --port {OpenMessagingBenchmarkWorkers.PORT} \
+                      --stats-port {OpenMessagingBenchmarkWorkers.STATS_PORT} \
+                      >> {OpenMessagingBenchmarkWorkers.STDOUT_STDERR_CAPTURE} 2>&1 & disown"
+
+        with node.account.monitor_log(OpenMessagingBenchmarkWorkers.
+                                      STDOUT_STDERR_CAPTURE) as monitor:
+            node.account.ssh(start_cmd)
+            monitor.wait_until(
+                "Javalin has started",
+                timeout_sec=20,
+                backoff_sec=4,
+                err_msg=
+                "Open Messaging Benchmark worker service didn't finish startup"
+            )
+        self.logger.info(
+            f"Open Messaging Benchmark worker is successfully started on node {node.account.hostname}"
+        )
+
+    def raise_on_bad_log_lines(self, node):
+        """
+        Check if benchmark worker logfile contains errors
+        """
+        bad_lines = collections.defaultdict(list)
+        self.logger.info(
+            f"Scanning node {node.account.hostname} log for errors...")
+
+        for line in node.account.ssh_capture(
+                f"grep -e TimeoutException {OpenMessagingBenchmarkWorkers.STDOUT_STDERR_CAPTURE} || true"
+        ):
+            if "No such file or directory" not in line:
+                bad_lines[node].append(line)
```

Comment:
what sort of line is this looking for? it looks like it is looking for a line that contains "TimeoutException" and that the same line also needs to contain "No such file or directory"?

Reply:
Same question for the other service.

Reply:
It should not contain "No such file or directory"

Reply:
oh I missed the `not`, thanks for point that out.

what is a line that contains `TimeoutException` and `No such file or directory` which you don't want to treat as a bad line?

it would be useful to explain what is being searched for in the logs. otherwise other developers will have to reverse engineer the analysis that you've done here.